ledger/src/blockstore_db.rs

```diff
@@ -305,8 +317,40 @@ impl Rocks {
             new_cf_descriptor::<BankHash>(&access_type, &oldest_slot),
             new_cf_descriptor::<Root>(&access_type, &oldest_slot),
             new_cf_descriptor::<Index>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredData>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot),
+            match options.shred_storage_type {
+                ShredStorageType::RocksLevel => {
+                    new_cf_descriptor::<ShredData>(&access_type, &oldest_slot)
+                }
+                ShredStorageType::RocksFifo => {
+                    if options.shred_data_cf_size > FIFO_WRITE_BUFFER_SIZE {
+                        new_cf_descriptor_fifo::<ShredData>(&options.shred_data_cf_size)
+                    } else {
+                        warn!(
+                            "shred_data_cf_size is must be greater than {} for RocksFifo.",
+                            FIFO_WRITE_BUFFER_SIZE
+                        );
+                        warn!("Fall back to ShredStorageType::RocksLevel for cf::ShredData.");
+                        new_cf_descriptor::<ShredData>(&access_type, &oldest_slot)
+                    }
+                }
+            },
+            match options.shred_storage_type {
+                ShredStorageType::RocksLevel => {
+                    new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot)
+                }
+                ShredStorageType::RocksFifo => {
+                    if options.shred_code_cf_size > FIFO_WRITE_BUFFER_SIZE {
+                        new_cf_descriptor_fifo::<ShredCode>(&options.shred_code_cf_size)
+                    } else {
+                        warn!(
+                            "shred_code_cf_size is must be greater than {} for RocksFifo.",
+                            FIFO_WRITE_BUFFER_SIZE
+                        );
+                        warn!("Fall back to ShredStorageType::RocksLevel for cf::ShredCode.");
+                        new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot)
+                    }
+                }
+            },
```

Comment:
Can we extract this into a common function that takes a generic `T` to distinguish between `ShredCode` and `ShredData`

Reply:
The code was on its stacked PR which is already landed.

I've created #23103 for this.